### PR TITLE
repairing heatmap updates for inner nodes

### PR DESCRIPTION
**Pull request by mattulbrich**
#3391 has been merged prematurely. This is an addition addressing a request made there by @WolframPfeifer .

All documentation from #3391 applies to here.

> @mattulbrich Do you have a good idea how to fix the problem with the non-goal views? If so, could you update it, please? If not, feel free to merge. The remaining UI bug is minor.

### CODE REVIEW

File: key.ui/src/main/java/de/uka/ilkd/key/gui/actions/HeatmapToggleAction.java [hunk 1]
```diff
@@ -67,5 +69,11 @@ public void actionPerformed(ActionEvent e) {
         // this updates the heatmap highlights
         mainWindow.getCurrentGoalView().getHighlighter().removeAllHighlights();
         mainWindow.getCurrentGoalView().printSequent();
+        Component component = mainWindow.getMainFrame().getContent();
```
Comment (WolframPfeifer) [inline key.ui/src/main/java/de/uka/ilkd/key/gui/actions/HeatmapToggleAction.java:72]: This should now always update twice, right? See my suggestion ...

File: key.ui/src/main/java/de/uka/ilkd/key/gui/nodeviews/MainFrame.java [hunk 2]
```diff
@@ -113,4 +113,8 @@ public boolean isShowTacletInfo() {
     public void scrollTo(int y) {
         scrollPane.getVerticalScrollBar().setValue(y);
     }
+
+    public Component getContent() {
```
Comment (WolframPfeifer) [inline key.ui/src/main/java/de/uka/ilkd/key/gui/nodeviews/MainFrame.java:117]: Component is always a SequentView. My suggestion changes the type to remove all the casts.

**Review by WolframPfeifer — APPROVED** [review]
Thanks for this. Somehow I did not see that the SequentView was stored in MainFrame.

The fix is fine. However, I attached a small patch with a refactoring as a suggestion that prevents double updates and some unnecessary casting in MainFrame. I would be happy if you could apply it, I cannot, since this is a PR from your repo.

[suggestion.txt](https://github.com/user-attachments/files/16785638/suggestion.txt)

File: key.ui/src/main/java/de/uka/ilkd/key/gui/actions/HeatmapToggleAction.java (hunk 1 above)
Comment (mattulbrich) [reply key.ui/src/main/java/de/uka/ilkd/key/gui/actions/HeatmapToggleAction.java:72]: Don't know. If it works with your suggestion, then ... yes. I was not sure. Thanks for finding out.

**Comment by mattulbrich** [comment]
checker framework is fixed on other branch.

**Pull request merged**